config/registry/default_schema.yaml

```diff
@@ -0,0 +1,719 @@
+registry:
+  schema:
+    version: 0
+    dataUnits:
+      Camera:
+        doc: >
+          A camera that defines a set of physical filters and sensors, and defines a
+          numbering system for its visits.
+        link:
+        - 
+          name: camera
+          type: string
+          foreign_key: Camera.camera
+        tables:
+          Camera:
+            doc: >
+              Definition table containing allowable values for e.g. Dataset.camera.
+            columns:
+            - 
+              name: camera
+              type: string
+              primary_key: true
+    
+      AbstractFilter:
+        doc: >
+          A filter that is not associated with a particular camera.  An abstract
+          filter can be used to relate similar physical filters, and is typically the
+          filter associated with coadds.
+        link:
+        - 
+          name: abstract_filter
+          type: string
+          foreign_key: AbstractFilter.abstract_filter
+        tables:
+          AbstractFilter:
+            doc: >
+              Definition table containing allowable values for e.g. Dataset.abstract_filter.
+            columns:
+            - 
+              name: abstract_filter
+              type: string
+              primary_key: true
+              doc: >
+                Name of the filter, typically a single letter (e.g. "r" or "i").
+    
+      PhysicalFilter:
+        dependencies:
+          required:
+          - Camera
+          optional:
+          - AbstractFilter
+        doc: >
+          A filter associated with a particular camera.  Physical filters are used to
+          identify datasets that can only be associated with a single observation.
+        link:
+        - 
+          name: physical_filter
+          type: string
+          foreign_key: PhysicalFilter.physical_filter
```

Comment:
Should this have one `link` (`physical_filter`) or two (`physical_filter` and `camera`)?  I think you have all of the necessary information here, but just to confirm it's structured:
 - No two `DataUnits` may have the same `link`.
 - A single `DataUnits` may (rarely) have more than one `link` (e.g. `ExposureRange`).
 - A `DataUnit` is fully specified by its `link` and the `link`s of all `required` `dependencies`.  If the `DataUnit` has a table, "fully specified" means those columns correspond to its (compound) primary key.

Reply:
Yes, that is correct. I am assuming that when a link is inserted the dependencies are first expanded and also included. Thus dupplicate links are not allowed.